tests/testautonatv2.nim

```diff
@@ -107,3 +119,32 @@ suite "AutonatV2":
 
     # DialBackResponse
     checkEncodeDecode(DialBackResponse(status: DialBackStatus.Ok))
+
+  asyncTest "Successful DialRequest":
+    let
+      src = newStandardSwitch()
+      dst = createAutonatV2Switch()
+    await src.start()
+    await dst.start()
+    defer:
+      await allFutures(src.stop(), dst.stop())
+
+    await src.connect(dst.peerInfo.peerId, dst.peerInfo.addrs)
+    let client = AutonatV2Client.new(newRng(), src)
+    let dialResp = (
+      await client.sendDialRequest(
+        src, dst.peerInfo.peerId, dst.peerInfo.addrs, src.peerInfo.addrs
+      )
+    ).get()
```

Comment:
issues pointed with comments related to `Opt` in return are shown here, code gets result like it's given.